src/tmpfiles/tmpfiles.c

```diff
@@ -857,13 +864,20 @@ static int path_set_perms(Item *i, const char *path) {
                 return r;
         }
 
-        if (fstat(fd, &st) < 0)
-                return log_error_errno(errno, "Failed to fstat() file %s: %m", path);
+        if (i->type == EMPTY_DIRECTORY) {
+                struct stat stbuf;
+
+                /* FIXME: introduce fd_is_dir() helper ? */
+                if (fstat(fd, &stbuf) < 0)
+                        return log_error_errno(errno, "Failed to fstat() file %s: %m", path);
 
-        if (i->type == EMPTY_DIRECTORY && !S_ISDIR(st.st_mode))
-                return log_error_errno(EEXIST, "'%s' already exists and is not a directory. ", path);
+                if (!S_ISDIR(stbuf.st_mode))
+                        return log_error_errno(EEXIST, "'%s' already exists and is not a directory. ", path);
 
-        return fd_set_perms(i, fd, &st);
+                st = &stbuf;
```

Comment:
uh, this is going to result in memory corruption... you need to define stbuf in the outer {}, otherwise you have the pointer "st" pointing to memory that is already released by the }...